packages/wrangler/src/queues/client.ts

```diff
@@ -46,208 +48,248 @@ export type Consumer = ScriptReference & {
 	type: string;
 };
 
-export interface QueueResponse {
-	queue_id: string;
+export interface TypedConsumerResponse extends Consumer {
 	queue_name: string;
 	created_on: string;
-	modified_on: string;
-	producers: ScriptReference[];
-	producers_total_count: number;
-	consumers: Consumer[];
-	consumers_total_count: number;
-	settings?: QueueSettings;
+}
+
+export interface PostTypedConsumerBody {
+	type: string;
+	script_name?: string;
+	environment_name?: string;
+	settings: ConsumerSettings;
+	dead_letter_queue?: string;
+}
+
+export interface ConsumerSettings {
+	batch_size?: number;
+	max_retries?: number;
+	max_wait_time_ms?: number;
+	max_concurrency?: number | null;
+	visibility_timeout_ms?: number;
+	retry_delay?: number;
+}
+
+export class QueuesUrlBuilder {
+	private url: string;
+	private hasQueueId: boolean;
+	constructor(accountId: string) {
+		this.url = `/accounts/${accountId}/queues`;
+		this.hasQueueId = false;
+	}
+
+	withQueueId(queueId: string): QueuesUrlBuilder {
+		this.url += `/${queueId}`;
+		this.hasQueueId = true;
+		return this;
+	}
+
+	withConsumer(consumerId?: string): QueuesUrlBuilder {
+		if (!this.hasQueueId) {
+			throw new UserError(
+				"Can't access consumers without a queueId specified."
```

Comment:
it is surprising to see a UserError thrown from the depths of a utility function - can this be removed? I think the UserError thrown higher up covers this case, right? `throw new UserError(`No http_pull consumer exists for queue ${queueName}`);`

Reply:
It might not hit that `UserError` on some code paths. But tbh this is an internal thing. I'm ok with it just forming a malformed URL, would get a 404/400 from the API.